upgrade_test.py

```diff
@@ -355,6 +387,10 @@ def test_rolling_upgrade(self):
 
         self.verify_stress_thread(entire_write_cs_thread_pool)
 
+        # run 'nodetool upgradesstables' on all nodes and check/wait for all file to be upgraded
```

Comment:
We already executed `nodetool upgradesstables` in each upgrade/rollback. Why do we repeatedly upgrade sstables here?

Reply:
me and @roydahan set over the logs, and seen that doing those upgrades while not all the cluster is upgraded, is no-op, i.e. there's no agreement yet in the cluster on the sstable format.

hence we decided to add this stage, to upgrade everything, and check everything is upgraded.

any idea on figuring what is the latest format version ?  that we expect ? maybe grepping the logs for the features ?

Reply:
Can we remove upgradesstables in each upgrade or rollback, and only upgrade them in the end?

The easy way to identify the format is checking sstable file name.
- la-...  old format
- mc-... new sst3 format

Reply:
I think it's good to cover the behavior of using this tool in cluster not fully upgraded, to make sure we don't leave the cluster in a state we can't rollback.

as for the filenames I know that, question is what is the highest version scylla support and if we have a way to query it without hardcoding it in the code, as I'm doing now

Reply:
sst3 format is supported from opensource 3.0, enterprise 2019.1

Reply:
A simple way is checking `test_sst3` parameter.
We only execute upgradesstables when we sst3 format is enabled.

Reply:
a question what's sst stands for ?

and still I was thinking on a better way to achieve get what is configured, without prior knowledge of which version support what

Reply:
@roydahan, 
what's you take on leaving only the `nodetool upgradesstables` after the whole cluster is upgraded ? or we are good with it as it is ?

Reply:
> a question what's sst stands for ?

sst -> SSTable
 
> and still I was thinking on a better way to achieve get what is configured, without prior knowledge of which version support what

Agree with you.

Reply:
see my last commit, trying to address this

Reply:
1. I think we should leave it as is, IOW test the upgradesstables on each step.
We aren't testing only happy path, we would like to test things that customers may do (like what me and Israel thought).

2. I'm against of using the yaml parameter test_sst3. we should automatically detect it.